test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -551,6 +566,130 @@ public void narrativeDecoration() throws Exception {
         NarrativeInfoDecoratorTest.compare(res.getAccessGroupNarrativeInfo(), expected);
     }
 
+    @Test
+    public void optionalAuthForSearch() throws Exception {
```

Comment:
This test is waaaaaaay too long, and, I presume, tests many assertions. It needs to be broken up into smaller tests, potentially with set up code reused in a method.

Another possibility is that if the unit tests for SearchMethods are complete, this test could probably be reduced to just testing regular and unauthed search integration (assuming it's now testing more than that).

Reply:
it is just testing regular and unauthed search though

Reply:
There's two tests right there, in that case - and why is it 120 lines of code do just do two tests?

Reply:
everything before the try catch is setting up the workspace and indexing objects

Reply:
try catch is doing the calls, and everything after is checking it got the right results. Uma also indents alot so it looks long

Reply:
What I'm trying to get at is that the test should be as easy as possible to read and understand - we just had this problem yesterday with a test neither of us could understand, so you should get where I'm coming from :p

To make tests easy to understand they should 
1) be as short as possible
2) have a name describing what being tested
2a) with extra documentation if necessary
3) Only test one thing or concept or whatever. There are some people that say each test should have a single assert statement, but I don't go that far. IMO each test should test one action - like one call to a method. That might take multiple asserts to test thoroughly, but the point is that one method call tested per test is a good general rule.

If there's a lot of set up code that's shared between two method call tests, put it it its own method with a descriptive name.

4) Favor unit tests over integration tests. They're faster, easier to understand, and easier to debug. Integration tests should just make sure the basic paths through the integrated objects work and the object's message passing works. Unit tests should exercise the bulk of the code.

Reply:
Let's not worry about trying to tighten this up further.  I really want to get this merged and running in CI so Erik can start testing it.

Reply:
It should take a few minutes to fix. We're already having this problem where apparently tests aren't preventing bugs from being introduced, or the tests we're adding aren't actually testing the code properly. The fix for that is tests that are clearer so it's easier to reason about them and figure out when they're not working.

Reply:
@MrCreosote I am not sure what I can take out here though that would make this easier to read. It follows the same pattern as all the other tests.

Reply:
You'd have separate tests for the 3 methods and 2 access modes, so 6 tests instead of one, each with descriptive names, and you'd put the setup code in a common method.